src/Serializer/ItemNormalizer.php

```diff
@@ -37,11 +39,17 @@ public function normalize($object, $format = null, array $context = [])
 
     /**
      * {@inheritdoc}
+     *
+     * @throws InvalidArgumentException
      */
     public function denormalize($data, $class, $format = null, array $context = [])
     {
         // Avoid issues with proxies if we populated the object
         if (isset($data['id']) && !isset($context['object_to_populate'])) {
+            if (isset($context['allow_update']) && true !== $context['allow_update']) {
```

Comment:
I also took a look to this and even if I agree on the general principle of name spacing things, I'm afraid that it's too late for this particular change in rc3.

It will break almost all apps already  using API Platform 2 for a very little benefit (problems are very unlikely to happen, especially because we also maintain the Symfony Serializer and we can prevent problems in Symfony core).

Prefixing context key will also make the config of all apps using API Platform uglier and harder to remain.

I'm for merging this PR as is.